rpc/src/rpc.rs

```diff
@@ -1015,28 +1000,30 @@ impl JsonRpcRequestProcessor {
                 self.check_status_is_complete(slot)?;
                 let result = self.blockstore.get_rooted_block(slot, true);
                 self.check_blockstore_root(&result, slot)?;
-                let configure_block = |versioned_block: VersionedConfirmedBlock| {
-                    let confirmed_block = versioned_block
-                        .into_legacy_block()
-                        .ok_or(RpcCustomError::UnsupportedTransactionVersion)?;
-                    let mut confirmed_block =
-                        confirmed_block.configure(encoding, transaction_details, show_rewards);
-                    if slot == 0 {
-                        confirmed_block.block_time = Some(self.genesis_creation_time());
-                        confirmed_block.block_height = Some(0);
-                    }
-                    Ok(confirmed_block)
-                };
+                let encode_block =
+                    |mut confirmed_block: ConfirmedBlock| -> Result<UiConfirmedBlock> {
+                        if slot == 0 {
+                            confirmed_block.block_time = Some(self.genesis_creation_time());
+                            confirmed_block.block_height = Some(0);
+                        }
+                        Ok(confirmed_block
+                            .encode_with_options(encoding, transaction_details, show_rewards)
+                            .map_err(RpcCustomError::from)?)
```

Comment:
Why switch the order here (requiring the passed-in block to be mut)?

Reply:
Undid this in the new PR